internal/ingestion/service/burst_backpressure_test.go

```
@@ -174,6 +175,39 @@ func TestProcessMessage_BurstNoTaskLossUnderBackpressure(t *testing.T) {
+// saturated channel still polls (and lets blocked sends drain) instead of
+// stalling until AckWait expires on in-flight messages.
+func TestFetchBudget_NeverExceedsChannelCapacity(t *testing.T) {
+	ingestor := newUnitIngestor("test", 2, []string{"pdf"}) // channel cap = 4
+	defer ingestor.Stop(context.Background())
+
+	if got := ingestor.fetchBudget(); got != 2 {
+		t.Fatalf("empty channel: fetchBudget = %d, want 2 (capped by maxConcurrency)", got)
+	}
+
+	for i := 0; i < cap(ingestor.taskChan)-1; i++ {
+		ingestor.taskChan <- taskpkg.NewTaskContextForScheduling(nil, &entity.IngestionTask{ID: "filler"})
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)